flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -262,7 +262,9 @@ public void prepareSnapshotPreBarrier(long checkpointId) throws Exception {
 		// go forward through the operator chain and tell each operator
 		// to prepare the checkpoint
 		for (StreamOperatorWrapper<?, ?> operatorWrapper : getAllOperators()) {
-			operatorWrapper.getStreamOperator().prepareSnapshotPreBarrier(checkpointId);
+			if (!operatorWrapper.isClosed()) {
+				operatorWrapper.getStreamOperator().prepareSnapshotPreBarrier(checkpointId);
+			}
```

Comment:
Maybe it would be better push the `isClosed()` checks into the wrapper class? And instead of:
```
if (!operatorWrapper.isClosed()) {
	operatorWrapper.getStreamOperator().prepareSnapshotPreBarrier(checkpointId);
}
```
have here (and in other places):
```
operatorWrapper.prepareSnapshotPreBarrier(checkpointId);
```
?

Reply:
I was also think about that. The question is should we then replace it for all `StreamOperatorWrapper.getOperator ` calls eventually?

Reply:
Maybe in the future?